### PR TITLE
[Docs] translate initialize

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/advanced_tutorials/initialize.md [hunk 1]
```diff
@@ -1,3 +1,325 @@
 # Initialization
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/initialize.html).
+Usually, we'll customize our module based on [nn.Module](https://pytorch.org/docs/stable/nn.html?highlight=nn%20module#module-torch.nn.modules), which is implemented by Native PyTorch. Also, [torch.nn.init](https://pytorch.org/docs/stable/nn.init.html?highlight=kaiming#torch.nn.init.kaiming_normal_) could help us initialize the parameters of the model easily. To simplify the process of model construction and initializationm, MMEngine design the [BaseModule](mmengine.model.BaseModule) to help us define and initialize the model from config easily.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:3]:
```suggestion
Usually, we'll customize our module based on [nn.Module](https://pytorch.org/docs/stable/nn.html?highlight=nn%20module#module-torch.nn.modules), which is implemented by Native PyTorch. Also, [torch.nn.init](https://pytorch.org/docs/stable/nn.init.html?highlight=kaiming#torch.nn.init.kaiming_normal_) could help us initialize the parameters of the model easily. To simplify the process of model construction and initialization, MMEngine designed the [BaseModule](mmengine.model.BaseModule) to help us define and initialize the model from config easily.
```

File: docs/en/advanced_tutorials/initialize.md [hunk 2]
```diff
@@ -1,3 +1,325 @@
...
+
+## Initialize the model from config
+
+The core function of `BaseModule` is that it could help us to initialize the model from config. Subclasses inherit from `BaseModule` could define the `init_cfg` in the `__init__` function, and we can choose the method of initialization by configuring `init_cfg`.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:7]:
```suggestion
The core function of `BaseModule` is that it could help us to initialize the model from config. Subclasses inherited from `BaseModule` could define the `init_cfg` in the `__init__` function, and we can choose the method of initialization by configuring `init_cfg`.
```

File: docs/en/advanced_tutorials/initialize.md [hunk 3]
```diff
@@ -1,3 +1,325 @@
...
+| [ConstantInit](../api.html#mmengine.model.ConstantInit)         |    Constant     | Initialize the weight and bias with a constant, commonly used for Convolution                                                            |
+| [XavierInit](../api.html#mmengine.model.XavierInit)             |     Xavier      | Initialize the weight by `Xavier` initialization, and initialize the bias with a constant                                                |
+| [NormalInit](../api.html#mmengine.model.NormalInit)             |     Normal      | Initialize the weight by normalize distribution, and initialize the bias with a constant                                                 |
+| [TruncNormalInit](../api.html#mmengine.model.TruncNormalInit)   |   TruncNormal   | Initialize the weight by truncated normalize distribution, and initialize the bias with a constant，commonly used for Transformer        |
+| [UniformInit](../api.html#mmengine.model.UniformInit)           |     Uniform     | Initialize the weight by union distribution, and initialize the bias with a constant，commonly used for convolution                      |
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:17]:
```suggestion
| [UniformInit](../api.html#mmengine.model.UniformInit)           |     Uniform     | Initialize the weight by uniform distribution, and initialize the bias with a constant，commonly used for convolution                      |
```

File: docs/en/advanced_tutorials/initialize.md [hunk 4]
```diff
@@ -1,3 +1,325 @@
...
+
+# Save the checkpoint.
+toy_net = ToyNet()
+torch.save(toy_net.state_dict(), './pretrained.pth')
+pretrained = './pretrained.pth'
+
+toy_net = ToyNet(init_cfg=dict(type='Pretrained', checkpoint=pretrained))
+```
+
+and then we can configure the `init_cfg` to make to load the pretrained model by calling `initi_weights()` after its construction.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:48]:
```suggestion
and then we can configure the `init_cfg` to make it load the pretrained model by calling `initi_weights()` after its construction.
```

File: docs/en/advanced_tutorials/initialize.md [hunk 5]
```diff
@@ -1,3 +1,325 @@
...
+```
+
+If `init_cfg` is a `dict`, `type` means a kind of initializer registered in `WEIGHT_INITIALIZERS`. The `Pretrained` means `PretrainedInit`, which could help us to load the target checkpoint.
+All initializers have the same mapping relationship like `Pretrained` -> `PretrainedInit`, which strips the suffix `Init` of the class name. The `checkpoint` argument of `PretrainedInit` means the path of the checkpoint. It could be a local path or a URL.
+
+### Commonly used initialization methods
+
+Similarly, we could use the `Kaiming` initialization just like `Pretrained` initializer. For example, we could make `init_cfg=dict(type='Kaiming', layer='Conv2d')` to initialize all `Conv2d` module with `Kaiming` initialization.
+
+Sometimes we need to initialize the model with different initialization methods for different modules. For example, we could initialize the `Conv2d` module with `Kaiming` initialization and initialize the `Linear` module with `Xavier` initialization. We could make `init_cfg=dict(type='Kaiming', layer='Conv2d', override=dict(type='Xavier', layer='Linear'))`:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:67]: Is it too early for `override` to occur here?

File: docs/en/advanced_tutorials/initialize.md [hunk 6]
```diff
@@ -1,3 +1,325 @@
...
+
+- Commonly used for the initialization of `Conv2d`, `Linear` and other underlying module. All initialization methods should be managed by `WEIGHT_INITIALIZERS`
+- Dynamic initialization controlled by `init_cfg`
+
+**2. Customize `init_weights`**
+
+- Compared to configuring the `init_cfg`, implementing the `init_weights` is simpler and does not require registration. However, it is not as flexible as `init_cfg`, and it is not possible to initialize the module dynamically.
+
+```{note}
+- The priorify of init_weights is higher `init_cfg`
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:275]:
```suggestion
- The priorify of init_weights is higher than `init_cfg`
```

File: docs/en/advanced_tutorials/initialize.md [hunk 7]
```diff
@@ -1,3 +1,325 @@
...
+- Runner will call `init_weights` in Runner.train()
+```
+
+### Ininitailize module with function
+
+As mentioned in prior \[section\](#Customize the initialization method), we could customize our initialization in `init_weights`. To make it more convenient initialize module, MMEngine provides a series of **module initialization functions** to initialize the whole module based on `torch.nn.init`. For example, we want to initialize the weights of the convolutional layer with normal distribution and initialize the bias of the convolutional layer with a constant. The implementation of `torch.nn.init` is as follows:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:281]:
```suggestion
As mentioned in prior [section](#Customize the initialization method), we could customize our initialization in `init_weights`. To make it more convenient to initialize modules, MMEngine provides a series of **module initialization functions** to initialize the whole module based on `torch.nn.init`. For example, we want to initialize the weights of the convolutional layer with normal distribution and initialize the bias of the convolutional layer with a constant. The implementation of `torch.nn.init` is as follows:
```

File: docs/en/advanced_tutorials/initialize.md [hunk 8]
```diff
@@ -1,3 +1,325 @@
...
+xavier_init(model)
+```
+
+Currently, MMEngine provide the following initialization function:
+
+| initialization function                                               | function                                                                                                                                 |
+| :-------------------------------------------------------------------- | :--------------------------------------------------------------------------------------------------------------------------------------- |
+| [constant_init](../api.html#mmengine.model.constant_init)             | Initialize the weight and bias with a constant, commonly used for Convolution                                                            |
+| [xavier_init](../api.html#mmengine.model.xavier_init)                 | Initialize the weight by `Xavier` initialization, and initialize the bias with a constant                                                |
+| [normal_init](../api.html#mmengine.model.normal_init)                 | Initialize the weight by normalize distribution, and initialize the bias with a constant                                                 |
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:320]:
```suggestion
| [normal_init](../api.html#mmengine.model.normal_init)                 | Initialize the weight by normal distribution, and initialize the bias with a constant                                                 |
```

File: docs/en/advanced_tutorials/initialize.md [hunk 9]
```diff
@@ -1,3 +1,325 @@
...
+| [trunc_normal_init](../api.html#mmengine.model.trunc_normal_init)     | Initialize the weight by truncated normalize distribution, and initialize the bias with a constant，commonly used for Transformer        |
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:321]:
```suggestion
| [trunc_normal_init](../api.html#mmengine.model.trunc_normal_init)     | Initialize the weight by truncated normal distribution, and initialize the bias with a constant，commonly used for Transformer        |
```

File: docs/en/advanced_tutorials/initialize.md [hunk 10]
```diff
@@ -1,3 +1,325 @@
...
+| [uniform_init](../api.html#mmengine.model.uniform_init)               | Initialize the weight by union distribution, and initialize the bias with a constant，commonly used for convolution                      |
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:322]:
```suggestion
| [uniform_init](../api.html#mmengine.model.uniform_init)               | Initialize the weight by uniform distribution, and initialize the bias with a constant，commonly used for convolution                      |
```

File: docs/en/advanced_tutorials/initialize.md [hunk 11]
```diff
@@ -1,3 +1,325 @@
...
+| [kaiming_init](../api.html#mmengine.model.kaiming_init)               | Initialize the weight by `Kaiming` initialization, and initialize the bias with a constant. Commonly used for convolution                |
+| [caffe2_xavier_init](../api.html#mmengine.model.caffe2_xavier_init)   | `Xavier` initialization in Caffe2, and `Kaiming` initialization in PyTorh with `fan_in` and `normal` mode. Commonly used for convolution |
+| [bias_init_with_prob](../api.html#mmengine.model.bias_init_with_prob) | Initialize the model with the pretrained model                                                                                           |
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/initialize.md:325]: Explanation seems wrong

File: docs/en/advanced_tutorials/initialize.md (hunk 5 above)
Comment (HAOCHENYE) [reply docs/en/advanced_tutorials/initialize.md:67]: Stupid me use copilot stupidly 😢 .

File: docs/en/advanced_tutorials/initialize.md [hunk 12]
```diff
@@ -1,3 +1,325 @@
...
+
+## Initialize the model from config
+
+The core function of `BaseModule` is that it could help us to initialize the model from config. Subclasses inherited from `BaseModule` could define the `init_cfg` in the `__init__` function, and we can choose the method of initialization by configuring `init_cfg`.
+
+Currently, we support the following initialization methods:
+
+| Initializer                                                     | Registered name | Function                                                                                                                                 |
+| :-------------------------------------------------------------- | :-------------: | :--------------------------------------------------------------------------------------------------------------------------------------- |
+| [ConstantInit](../api.html#mmengine.model.ConstantInit)         |    Constant     | Initialize the weight and bias with a constant, commonly used for Convolution                                                            |
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/initialize.md:13]: links failed

File: docs/en/advanced_tutorials/initialize.md [hunk 13]
```diff
@@ -1,3 +1,325 @@
...
+KaimingInit: a=0, mode=fan_out, nonlinearity=relu, distribution =normal, bias=0
+
+08/19 16:50:24 - mmengine - INFO -
+conv.bias - torch.Size([1]):
+KaimingInit: a=0, mode=fan_out, nonlinearity=relu, distribution =normal, bias=0
+```
+
+### More fine-grained initialization
+
+Sometimes we need to initialize the same type of module with different type of initialization. For example, we've defined `conv1` and `conv2` submodules, and we want to initialize the `conv1` with `Kaiming` initialization and initialize the `conv2` with `Xavier` initialization. We could configure the init_cfg with `override`:
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/initialize.md:138]:
```suggestion
Sometimes we need to initialize the same type of module with a different types of initialization. For example, we've defined `conv1` and `conv2` submodules, and we want to initialize the `conv1` with `Kaiming` initialization and initialize the `conv2` with `Xavier` initialization. We could configure the init_cfg with `override`:
```

**Review by zhouzaida — APPROVED** [review]
